npa.js

```diff
@@ -18,6 +18,17 @@ const isFilename = /[.](?:tgz|tar.gz|tar)$/i
 function npa (arg, where) {
   let name
   let spec
+  if (arg && arg.toString().match(/^\[object/)) {
```

Comment:
why not do `instanceof Result` here?

Reply:
It might be a different `npm-package-arg` pkg in the tree. 🙃 

But I guess in that case, it's still best for us to make a new instance anyway?